ccmlib/scylla_node.py

```diff
@@ -1337,6 +1338,34 @@ def upgrade(self, upgrade_to_version):
     def rollback(self, upgrade_to_version):
         self.upgrader.upgrade(upgrade_version=upgrade_to_version, recover_system_tables=True)
 
+    def watch_rest_for_alive(self, nodes, timeout=120):
+        """
+        Use the REST API to wait until this node detects that the nodes listed
+        in "nodes" become fully operational (live and no longer "joining").
+        This is similar to watch_log_for_alive but uses ScyllaDB's REST API
+        instead of the log file and waits for the node to be really useable,
+        not just "UP" (see issue #461)
+        """
+        tofind = nodes if isinstance(nodes, list) else [nodes]
+        tofind = set([node.address() for node in tofind])
+        url_live = f"http://{self.address()}:10000/gossiper/endpoint/live"
+        url_joining = f"http://{self.address()}:10000/storage_service/nodes/joining"
+        endtime = time.time() + timeout
+        while time.time() < endtime:
+            live = {}
```

Comment:
live should be initialized as `live = set()`
otherwise there might be a minuscule chance that `live - set(response.json())` below will return the following error: `TypeError: unsupported operand type(s) for -: 'dict' and 'set'`

Reply:
Oops, I forgot that {} wasn't an empty set :-(
Will fix this.